lib/config-util.js

```diff
@@ -115,6 +115,12 @@ Config.buildYargs = function (cwd) {
       describe: 'cache babel transpilation results for improved performance',
       global: false
     })
+    .option('es-modules', {
+      default: true,
```

Comment:
Is there any reason to not default this to `false`, I guess I'm not fully understanding what this flag buys us -- since ES-modules aren't supported in vanilla JS yet.

Reply:
I set the default true to match the existing behavior.  The biggest effect is that it puts the Babel parser into strict mode.

I can default to false if you prefer?

Reply:
If it makes us less sensitive, without sacrificing our ability to instrument anything, my temptation would be to default it to `false` ... what do you think?

perhaps someone from the babel project can chime in, CC: @loganfsmyth

Reply:
A couple relevant details:
1. The istanbul-lib-instrument default for esModules is false.
2. This is translated into the babelrc option sourcetype `'module'` or `'script'`.  The default for babel 6 and 7 is `sourceType: 'module'`.

So honestly I think `istanbul-lib-instrument` should also have `esModules: true` by default so that not specifying any option causes babel to behave as normal.  I also understand the argument for being less restrictive, I don't really see this as a huge issue either way.

https://babeljs.io/docs/en/babel-core/#options
https://babeljs.io/docs/en/next/babel-core#options

Reply:
Personally I think the JS ecosystem is in a bit of an unfortunate place here, since there's simply no way to 100% guarantee that you have the type of a file right without additional data. That said, in the context of `nyc`, and now that #863 has landed, it probably makes the most sense for `nyc` to default to `unambiguous` which you'll see in [the `next` docs]( babeljs.io/docs/en/next/babel-core#options). It essentially means that the parser will guess the type.

Does Istanbul actually use the strictness of code the do anything during instrumentation? I'm assuming it doesn't. If that's the case, guessing wrong doesn't actually affect anything in Istanbul and just makes things more likely to parse.

Reply:
@bcoe Set to false and updated the tests.